udata/migrations/2016-11-08-truncate-tags.js

```diff
@@ -0,0 +1,42 @@
+/*
+ * Ensure all tags have a correct length
+ *
+ */
+
+const MIN_TAG_LENGTH = 3
+const MAX_TAG_LENGTH = 32
+
+function migrate(collection) {
+
+    var objects = collection.find({tags: {$gt: []}})
+    var name = collection.getName()
+
+    print(name + ' candidates : ' + objects.count());
+
+    var objectsCounter = 0;
+    var tagsCounter = 0;
+
+    objects.forEach(function(object) {
+        var tags = object.tags.map(function(tag) {
+            if (tag.length > MAX_TAG_LENGTH)  {
+                tagsCounter++;
+                return tag.substring(0, MAX_TAG_LENGTH)
+            }
+            if (tag.length < MIN_TAG_LENGTH)  {
+                tagsCounter++;
+                return null
+            }
+            return tag;
+        });
+        tags = tags.filter(function(tag) { return tag !== null })
```

Comment:
Not necessary but just so you know: Mongo 3.2 support ES6 arrow functions so you can write:
```javascript
tags = tags.filter(tag => tag)  // Works because filter cast as boolean
```